pkg/scheduler/core/generic_scheduler.go

```diff
@@ -547,10 +539,13 @@ func podFitsOnNode(
 							} else {
 								predicateResults[predicateKey] = HostPredicate{Fit: fit, FailReasons: reasons}
 							}
-							result := predicateResults[predicateKey]
-							ecache.UpdateCachedPredicateItem(
-								pod.GetName(), info.Node().GetName(),
-								predicateKey, result.Fit, result.FailReasons, equivCacheInfo.hash, false)
+							// Skip update if NodeInfo is stale.
+							if cache != nil && cache.IsUpToDate(info) {
```

Comment:
Thanks! Just took a first round of review, I think this actually consistent with this comment: https://github.com/kubernetes/kubernetes/issues/61512#issuecomment-377577199, which in my opinion, should solve the problem. 

I need to check more edge cases tomorrow.

Reply:
Yes. Luckily we already had a generation number which seemed reasonable to use.